Debug-custom-ESLint-check-tests.md

```diff
@@ -0,0 +1,101 @@
+## Table of contents
+
+* [Introduction](#introduction)
+* [Print statements](#print-statements)
+* [Inspect the AST](#inspect-the-ast)
+
+## Introduction
+
+Especially for complex lint rules, it can be hard to figure out why a test failed. This page will give you some tips to help you debug your failing tests.
```

Comment:
This intro is a bit sudden, and perhaps more context can be given (e.g. that we're talking specifically about ESlint rules, etc.). Maybe start with something like "Sometimes, you may run into issues when ..." so that the reader can understand whether this page applies to them or not.